org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/util/configuration/FixedRequestAttributesBuilder.java

```diff
@@ -100,4 +102,15 @@ public static FixedRequestAttributesBuilder fromWorkspaceSettings(File projectDi
                 .buildScansEnabled(configuration.isBuildScansEnabled())
                 .arguments(CorePlugin.invocationCustomizer().getExtraArguments());
      }
+
+    public static FixedRequestAttributesBuilder fromProjectSettings(IProject project) {
+        WorkspaceConfiguration workspaceConfig = CorePlugin.workspaceConfigurationManager().loadWorkspaceConfiguration();
```

Comment:
We should delegate to `fromWorkspaceSettings` here and then just call the methods that are relevant for project settings.